modules/environment/login/login-inner.nix

```diff
@@ -0,0 +1,93 @@
+# Licensed under GNU Lesser General Public License v3 or later, see COPYING.
+# Copyright (c) 2019 Alexander Sosedkin and other contributors, see AUTHORS.
+
+{ config, lib, customPkgs, writeText }:
+
+let
+  inherit (customPkgs.packageInfo) cacert coreutils nix;
+in
+
+writeText "login-inner" ''
+  # This file is generated by nix-on-droid. DO NOT EDIT.
+
+  set -eu -o pipefail
+
+  [ "$#" -gt 0 ] || echo "Welcome to Nix-on-Droid!"
+
+  [ "$#" -gt 0 ] || echo "If nothing works, use the rescue shell and read ${config.build.installationDir}/usr/lib/login-inner"
+  [ "$#" -gt 0 ] || echo "If it does not help, report bugs at https://github.com/t184256/nix-on-droid-bootstrap/issues"
+
+  export USER="${config.user.userName}"
+  export HOME="${config.user.home}"
+
+  export GC_NPROCS=1  # to prevent gc warnings of nix, see https://github.com/NixOS/nix/issues/3237
+
+  ${lib.optionalString config.build.initialBuild ''
+    if [[ -e /etc/UNINTIALISED ]]; then
+      echo "Set default user profile..."
+      ${nix}/bin/nix-env --switch-profile /nix/var/nix/profiles/per-user/$USER/profile
+
+      [ "$#" -gt 0 ] || echo "Sourcing Nix environment..."
+      . ${nix}/etc/profile.d/nix.sh
+
+      export NIX_SSL_CERT_FILE=${cacert}
+
+      if [ -r ${config.build.installationDir}/proot.closure ]; then
+        echo "Injecting bundled talloc closure..."
+        ${nix}/bin/nix-store --import < ${config.build.installationDir}/talloc.closure
+        ${coreutils}/bin/rm -f ${config.build.installationDir}/talloc.closure
+        echo "Injecting bundled proot closure..."
+        ${nix}/bin/nix-store --import < ${config.build.installationDir}/proot.closure
+        ${coreutils}/bin/rm -f ${config.build.installationDir}/proot.closure
+      else
+        echo "No bundled talloc/proot closures found, which is OK for official builds..."
```

Comment:
For debugging purposes this may be helpful, but I prefer not to print this to a normal user

Reply:
I concede, this isn't particularly helpful.